charm_test.go

```diff
@@ -7,34 +7,33 @@ import (
 	"bytes"
 	"io"
 	"io/ioutil"
+	"os"
 	"path/filepath"
 	stdtesting "testing"
 
 	jc "github.com/juju/testing/checkers"
+	"github.com/juju/utils/fs"
 	gc "gopkg.in/check.v1"
 	"gopkg.in/yaml.v1"
 
-	"gopkg.in/juju/charm.v5"
-	charmtesting "gopkg.in/juju/charm.v5/testing"
+	"gopkg.in/juju/charm.v6-unstable"
 )
 
 func Test(t *stdtesting.T) {
 	gc.TestingT(t)
 }
 
-var TestCharms = charmtesting.NewRepo("internal/test-charm-repo", "quantal")
-
 type CharmSuite struct{}
 
 var _ = gc.Suite(&CharmSuite{})
 
 func (s *CharmSuite) TestReadCharm(c *gc.C) {
-	bPath := TestCharms.CharmArchivePath(c.MkDir(), "dummy")
-	ch, err := charm.ReadCharm(bPath)
+	ch, err := charm.ReadCharm(charmDirPath(c, "dummy"))
 	c.Assert(err, gc.IsNil)
 	c.Assert(ch.Meta().Name, gc.Equals, "dummy")
-	dPath := TestCharms.CharmDirPath("dummy")
-	ch, err = charm.ReadCharm(dPath)
+
+	bPath := archivePath(c, readCharmDir(c, "dummy"))
```

Comment:
Can we do "readCharmDir" internally in archivePath? it would look better: archivePath(c, "dummy"). If you want to leave archivePath "open", you could introduce a wrapper function archivePathWithDir(c, "dummy") maybe?